source/base/function_parser.cc

```diff
@@ -158,27 +158,25 @@ FunctionParser<dim>::init_muparser() const
   // check that we have not already initialized the parser on the
   // current thread, i.e., that the current function is only called
   // once per thread
-  Assert(fp.get().size() == 0, ExcInternalError());
+  ParserData &data = parser_data.get();
+  Assert(data.parsers.size() == 0 && data.vars.size() == 0, ExcInternalError());
 
-  // initialize the objects for the current thread (fp.get() and
-  // vars.get())
-  fp.get().reserve(this->n_components);
-  vars.get().resize(var_names.size());
+  // initialize the objects for the current thread
+  data.parsers.reserve(this->n_components);
+  data.vars.resize(var_names.size());
   for (unsigned int component = 0; component < this->n_components; ++component)
     {
-      fp.get().emplace_back(
-        std::make_unique<internal::FunctionParserImplementation::Parser>());
+      data.parsers.emplace_back(
+        new internal::FunctionParserImplementation::Parser());
       mu::Parser &parser =
         dynamic_cast<internal::FunctionParserImplementation::Parser &>(
-          *fp.get().back());
+          *data.parsers.back());
 
       for (const auto &constant : constants)
-        {
-          parser.DefineConst(constant.first, constant.second);
-        }
+        parser.DefineConst(constant.first, constant.second);
 
       for (unsigned int iv = 0; iv < var_names.size(); ++iv)
-        parser.DefineVar(var_names[iv], &vars.get()[iv]);
+        parser.DefineVar(var_names[iv], &data.vars[iv]);
```

Comment:
What happened to the `vars` variable? I must be missing where you've removed its declaration.

Reply:
`vars` is just the other thread-local array stored by `ParserData` - i.e., it was managed by the other `ThreadLocalStorage` object.